pkg/apis/autoscaling/validation/validation.go

```diff
@@ -165,6 +175,73 @@ func validateMetrics(metrics []autoscaling.MetricSpec, fldPath *field.Path, minR
 	return allErrs
 }
 
+func validateBehavior(behavior *autoscaling.HorizontalPodAutoscalerBehavior, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if behavior != nil {
+		if scaleUpErrs := validateScalingRules(behavior.ScaleUp, fldPath.Child("scaleUp")); len(scaleUpErrs) > 0 {
+			allErrs = append(allErrs, scaleUpErrs...)
+		}
+		if scaleDownErrs := validateScalingRules(behavior.ScaleDown, fldPath.Child("scaleDown")); len(scaleDownErrs) > 0 {
+			allErrs = append(allErrs, scaleDownErrs...)
+		}
+	}
+	return allErrs
+}
+
+var validSelectPolicyTypes = sets.NewString(string(autoscaling.MaxPolicySelect), string(autoscaling.MinPolicySelect), string(autoscaling.DisabledPolicySelect))
+var validSelectPolicyTypesList = validSelectPolicyTypes.List()
+
+func validateScalingRules(rules *autoscaling.HPAScalingRules, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if rules != nil {
+		if rules.StabilizationWindowSeconds != nil && *rules.StabilizationWindowSeconds < 0 {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("stabilizationWindowSeconds"), rules.StabilizationWindowSeconds, "must be greater than or equal to zero"))
+		}
+		if rules.StabilizationWindowSeconds != nil && *rules.StabilizationWindowSeconds > MaxStabilizationWindowSeconds {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("stabilizationWindowSeconds"), rules.StabilizationWindowSeconds,
+				fmt.Sprintf("must be less than or equal to %v", MaxStabilizationWindowSeconds)))
+		}
+		if rules.SelectPolicy != nil &&
+			*rules.SelectPolicy != autoscaling.MaxPolicySelect &&
```

Comment:
This can be simplified to `validSelectPolicyTypes.Has(*rules.SelectPolicy)` I think

Reply:
Acceptable as a followup PR if there's nothing else to block this